lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -416,39 +383,45 @@ computeTxSize proxy pm witTag action cs =
         dummyOutput :: Coin -> TxOut
         dummyOutput = TxOut $ Address $ BS.pack (1:replicate 56 0)
 
-        dummyKeyHash = SL.KeyHash . Hash.UnsafeHash $ dummyKeyHashRaw
+        dummyStakeCred = toCardanoStakeCredential
+            $ ChimericAccount dummyKeyHashRaw
+
+        dummyPoolId :: Cardano.PoolId
+        dummyPoolId = fromMaybe (error "dummyPoolId couldn't be constructed")
+            $ Cardano.deserialiseFromRawBytes (Cardano.AsHash Cardano.AsStakePoolKey)
+            $ BS.pack $ replicate 32 0
 
         certs = case action of
             Nothing -> []
             Just RegisterKeyAndJoin{} ->
-                [ Cardano.shelleyRegisterStakingAddress dummyKeyHash
-                , Cardano.shelleyDelegateStake dummyKeyHash dummyKeyHash
+                [ Cardano.makeStakeAddressRegistrationCertificate dummyStakeCred
+                , Cardano.makeStakeAddressDelegationCertificate dummyStakeCred dummyPoolId
                 ]
             Just Join{} ->
-                [ Cardano.shelleyDelegateStake dummyKeyHash dummyKeyHash
+                [ Cardano.makeStakeAddressDelegationCertificate dummyStakeCred dummyPoolId
                 ]
             Just Quit ->
-                [ Cardano.shelleyDeregisterStakingAddress dummyKeyHash
+                [ Cardano.makeStakeAddressDeregistrationCertificate dummyStakeCred
                 ]
 
-    dummyKeyHashRaw = BS.pack (replicate 28 0)
+    dummyKeyHashRaw = BA.convert $ BS.pack (replicate 28 0)
 
     wdrls = mkWithdrawals
-        proxy
+        networkId
         (ChimericAccount dummyKeyHashRaw)
         (withdrawal cs)
 
     (addrWits, certWits) =
-        ( Set.union
-            (Set.map dummyWitnessUniq $ Set.fromList (fst <$> CS.inputs cs))
-            (if Map.null wdrls then Set.empty else Set.singleton (dummyWitness "0"))
+        (
+            map dummyWitnessUniq (fst <$> CS.inputs cs)
+            <> [dummyWitness "0" | null wdrls]
```

Comment:
```hs
>>> [14|True]
[14]

>>> [14|False]
[]
```

This creates a dummy witness if the witness set is empty. Which is, the opposite of what we want :grimacing: ... The list comprehension is shorter / somewhat more elegant but also more confusing, so I'd rather put back the good old `if null then [] else [...]` which is less error-prone.

Reply:
Sorry, hlint asked for it.

Reply:
hlint is stupid. We are not (arguably, but I will argue anytime). Sometimes, when the suggestion isn't sound, we might as well turn it off. I reverted this to a more readable / less error-prone version.